Spigot-Server-Patches/0596-added-PlayerEntityReceiveEvent.patch

```diff
@@ -0,0 +1,69 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: ysl3000 <yannicklamprecht@live.de>
+Date: Mon, 12 Oct 2020 23:11:41 +0200
+Subject: [PATCH] added PlayerEntityReceiveEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+index 4efc40c01ec12b80bd7cf9d35cf0ea0df973baf7..0a6c42688b979bf7abce7744e70f502a8b41360b 100644
+--- a/src/main/java/net/minecraft/server/EntityTrackerEntry.java
++++ b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+@@ -17,6 +17,10 @@ import org.bukkit.entity.Player;
+ import org.bukkit.event.player.PlayerVelocityEvent;
+ // CraftBukkit end
+ 
++// Paper  start
++import org.bukkit.craftbukkit.event.CraftEventFactory;
++import org.bukkit.event.packet.PlayerEntityReceiveEvent;
++// Paper  end
+ public class EntityTrackerEntry {
+ 
+     private static final Logger LOGGER = LogManager.getLogger();
+@@ -282,7 +286,16 @@ public class EntityTrackerEntry {
+         Packet<?> packet = this.tracker.P();
+ 
+         this.headYaw = MathHelper.d(this.tracker.getHeadRotation() * 256.0F / 360.0F);
+-        consumer.accept(packet);
++        // Paper start
++        if(PlayerEntityReceiveEvent.getHandlerList().getRegisteredListeners().length > 0){
++            PlayerEntityReceiveEvent playerEntityReceiveEvent = CraftEventFactory.callPlayerEntityReceiveEvent(this.tracker, entityplayer);
++            if (!playerEntityReceiveEvent.isCancelled()) {
```

Comment:
could prolly replace this and the event factor change with
```suggestion
+            if (!new PlayerChunkReceiveEvent(this.tracker, entityplayer.getBukkitEntity()).callEvent()) {
```

Reply:
added